docker/irods_clients/ubuntu/16.04/Dockerfile

```diff
@@ -0,0 +1,169 @@
+# syntax=docker/dockerfile:1
+
+ARG BASE_IMAGE=ubuntu:16.04
+FROM $BASE_IMAGE as installer
+
+ARG IRODS_VERSION="4.2.7"
+ARG BATON_VERSION="4.1.0"
+ARG HTSLIB_VERSION="1.18"
+ARG SAMTOOLS_VERSION="1.18"
+ARG BCFTOOLS_VERSION="1.18"
+ARG HTSLIB_PLUGINS_VERSION="201712"
+
+COPY . /opt/docker/irods_clients
+
+RUN echo 'debconf debconf/frontend select Noninteractive' | debconf-set-selections && \
+    apt-get update && \
+    apt-get install -q -y --no-install-recommends \
+    apt-transport-https \
+    apt-utils \
+    ca-certificates \
+    curl \
+    dirmngr \
+    gnupg \
+    gnupg-agent \
+    lsb-release \
+    locales && \
+    locale-gen en_GB en_GB.UTF-8 && \
+    localedef -i en_GB -c -f UTF-8 -A /usr/share/locale/locale.alias en_GB.UTF-8
+
+ENV LANG=en_GB.UTF-8 \
+    LANGUAGE=en_GB \
+    LC_ALL=en_GB.UTF-8
+
+RUN curl -sSL https://packages.irods.org/irods-signing-key.asc | apt-key add - && \
+    echo "deb [arch=amd64] https://packages.irods.org/apt/ $(lsb_release -sc) main" |\
+    tee /etc/apt/sources.list.d/renci-irods.list && \
+    apt-get update && \
+    apt-get install -q -y --no-install-recommends \
+    irods-dev="${IRODS_VERSION}" \
+    irods-runtime="${IRODS_VERSION}" \
+    irods-icommands="${IRODS_VERSION}"
+
+RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys "E1DD270288B4E6030699E45FA1715D88E1DF1F24" && \
+    echo "deb https://ppa.launchpadcontent.net/git-core/ppa/ubuntu $(lsb_release -sc) main" |\
+    tee /etc/apt/sources.list.d/git-core.list && \
+    apt-get update && \
+    apt-get install -q -y --no-install-recommends \
+    git
+
+# libdeflate-dev is not available on Ubuntu 16.04
```

Comment:
~Thanks for highlighting this: can we download the libdeflate source, compile and install it then? I think this would be particularly important with I guess an old (relatively inefficient) zlib. Our SAM team colleagues recommend against compiling without libdeflate.~ So... fine for use in tests as this is intended - just not production which we're not going to use this image for...